src/device.ts

```diff
@@ -312,15 +312,24 @@ export class Device extends vscode.Disposable {
      */
     public async mkdir_p(path: string): Promise<void> {
         const names = path.split('/');
+
+        // If path had a preceding slash, remove the empty path segment
+        if(!names[0]) {
```

Comment:
nit: space after if

There is really only one rule about code style that matters and is used nearly universally and that is to be consistent and match the style around the code you are adding.

When I nit pick about code style, what I'm really saying is "By the way, this is the code style that is already being used in this project And since I'm asking you to make other changes anyway, could you please clean up these small things. I know its not really super important, but let's try to keep things neat and consistent if we can."

This kind of request is so common and so universal across projects I have contributed to, it has become second nature to me and I don't really think about it or feel the need to write out the long explanation. But, I'm guessing from our recent conversation that perhaps an explanation is needed this one time.

Reply:
I shall award myself half marks for the space on line 327.